packages/2d/src/components/Polygon.ts

```diff
@@ -0,0 +1,104 @@
+import {Shape, ShapeProps} from './Shape';
+import {initial, signal} from '../decorators';
+import {SignalValue, SimpleSignal} from '@motion-canvas/core/lib/signals';
+
+export interface PolygonProps extends ShapeProps {
+  /**
+   * {@inheritDoc Polygon.nSides}
+   */
+  nSides?: SignalValue<number>;
+
+  /**
+   * {@inheritDoc Polygon.angle}
+   */
+  angle?: SignalValue<number>;
+}
+
+export class Polygon extends Shape {
+  /**
+   * Sets the number of sides of the polygon. For example, a value of 6
+   * creates a hexagon.
+   *
+   * @remarks
+   * Note that the polygon is inscribed in a circle defined by the height
+   * and width. If height and width are unequal, the polygon is inscribed
+   * in the resulting ellipse.
+   *
+   * Since the polygon is inscribed in the circle, the actual displayed
+   * height and width may differ somewhat from the bounding rectangle. This
+   * will be particularly noticable if the number of sides is low, e.g. for a
+   * triangle.
+   *
+   *
+   * @example
+   * ```tsx
+   * <Polygon
+   *   x={0}
+   *   y={0}
+   *   width={320}
+   *   height={320}
+   *   nSides={6}
+   *   stroke={'#fff'}
+   *   lineWidth={8}
+   *   fill={'lightseagreen'}
+   * />
+   * ```
+   */
+  @initial(6)
+  @signal()
+  public declare readonly nSides: SimpleSignal<number, this>;
+
+  /**
+   * Changes the starting angle (in degrees) of the polygon.
+   *
+   * @remarks
+   * Use this to change the apparent rotation of the polygon.
+   *
+   * @example
+   * ```tsx
+   * <Polygon
+   *   x={0}
+   *   y={0}
+   *   width={320}
+   *   height={320}
+   *   nSides={6}
+   *   angle={30}
+   *   stroke={'#fff'}
+   *   lineWidth={8}
+   *   fill={'lightseagreen'}
+   * />
+   * ```
+   */
+  @initial(0)
+  @signal()
+  public declare readonly angle: SimpleSignal<number, this>;
+
+  public constructor(props: PolygonProps) {
+    super(props);
+  }
+
+  protected override getPath(): Path2D {
```

Comment:
`Shape`s should also implement a `getRipplePath()` method that generates the same path but expanded by `this.rippleSize()`. This is used for ripple effects:
```ts
yield* shape().ripple();
```

Reply:
Let me investigate.

Reply:
I have `ripple()` working.  Will prepare a fresh commit.